lib/elixir/test/elixir/enum_test.exs

```diff
@@ -407,14 +407,66 @@ defmodule EnumTest do
     assert Enum.intersperse([1, 2, 3], true) == [1, true, 2, true, 3]
   end
 
-  test "into/2" do
+  setup do
+    %{item: 1, date: ~D[2015-01-01], map: %{a: 1}, tuple: {:a, 1}}
+  end
+
+  test "into/2", c do
     assert Enum.into([a: 1, b: 2], %{}) == %{a: 1, b: 2}
     assert Enum.into([a: 1, b: 2], %{c: 3}) == %{a: 1, b: 2, c: 3}
     assert Enum.into(MapSet.new(a: 1, b: 2), %{}) == %{a: 1, b: 2}
     assert Enum.into(MapSet.new(a: 1, b: 2), %{c: 3}) == %{a: 1, b: 2, c: 3}
     assert Enum.into(%{a: 1, b: 2}, []) == [a: 1, b: 2]
     assert Enum.into(1..3, []) == [1, 2, 3]
     assert Enum.into(["H", "i"], "") == "Hi"
```

Comment:
Please don't use `setup` here. It will add extra work to all of our tests. Instead, do this:

```suggestion
  test "into/2" do
    assert Enum.into([a: 1, b: 2], %{}) == %{a: 1, b: 2}
    assert Enum.into([a: 1, b: 2], %{c: 3}) == %{a: 1, b: 2, c: 3}
    assert Enum.into(MapSet.new(a: 1, b: 2), %{}) == %{a: 1, b: 2}
    assert Enum.into(MapSet.new(a: 1, b: 2), %{c: 3}) == %{a: 1, b: 2, c: 3}
    assert Enum.into(%{a: 1, b: 2}, []) == [a: 1, b: 2]
    assert Enum.into(1..3, []) == [1, 2, 3]
    assert Enum.into(["H", "i"], "") == "Hi"
  end

  test "into/2 exceptions" do
    item = 1
    date = ~D[2015-01-01]
    map = %{a: 1}
    tuple = {:a, 1}
```